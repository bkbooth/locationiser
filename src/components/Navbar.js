import React from 'react';
import { Link as RouterLink, withRouter } from 'react-router-dom';
import styled from 'styled-components/macro';
import { useAuth } from '../components/Auth';
import { WhiteButton } from '../components/styles/Button';
import worldMapEmoji from '../assets/world-map-emoji.png';

const Header = styled.header`
  position: relative;
  display: flex;
  align-items: center;
  justify-content: space-between;
  background: ${({ theme }) => theme.colours.primary['500']};
  color: ${({ theme }) => theme.colours.shade['900']};
  padding: ${({ theme }) => theme.sizes.md};
  box-shadow: ${({ theme }) => theme.sizes.nil} ${({ theme }) => theme.sizes.xs}
    ${({ theme }) => theme.sizes.sm} rgba(0, 0, 0, 0.4);
  z-index: 10;
`;

const Title = styled.h1`
  display: flex;
  align-items: center;
  margin: ${({ theme }) => theme.sizes.nil};
  font-size: 1.7rem;
  letter-spacing: 0.05rem;
`;

const Link = styled(RouterLink)`
  color: ${({ theme }) => theme.colours.shade['900']};
  text-decoration: none;
  transition: opacity ${({ theme }) => theme.speeds.fast};

  &:hover,
  &:focus {
    opacity: 0.8;
    outline: none;
  }
`;

const Logo = styled.img`
  width: ${({ theme }) => theme.sizes.xl};
  height: auto;
  margin-right: ${({ theme }) => theme.sizes.sm};
`;

function Navbar({ history }) {
  const auth = useAuth();

  function handleLogin() {
    history.push('/login');
  }

  function handleLogout() {
    auth.handleLogout();
    history.push('/login');
  }

  return (
    <Header>
      <Link to="/">
        <Title>
          <Logo src={worldMapEmoji} alt="world map emoji" />
          locations
        </Title>
      </Link>
      <nav>
        {auth.isAuthenticated ? (
          <WhiteButton onClick={handleLogout} disabled={auth.isLoading}>
            Logout
          </WhiteButton>
        ) : (
          <WhiteButton onClick={handleLogin} disabled={auth.isLoading}>
            Login
          </WhiteButton>
        )}
      </nav>
    </Header>
  );
}

export default withRouter(Navbar);
