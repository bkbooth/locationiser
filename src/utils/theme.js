export const theme = {
  colours: {
    primary: {
      '500': '#0094c0',
      '600': '#007192',
    },
    shade: {
      '100': '#101010',
      '400': '#575757',
      '500': '#656565',
      '600': '#929292',
      '800': '#ececec',
      '900': '#fbfbfb',
    },
    positive: {
      '500': '#38c000',
      '600': '#2c9600',
    },
    negative: {
      '500': '#c00000',
    },
  },
  fonts: {
    sans: `-apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen-Sans, Ubuntu, Cantarell, 'Helvetica Neue', sans-serif`,
  },
  sizes: {
    nil: '0',
    xxs: '1px',
    xs: '5px',
    sm: '10px',
    md: '20px',
    lg: '30px',
    xl: '40px',
  },
  speeds: {
    fast: '0.2s',
    slow: '0.5s',
  },
};
