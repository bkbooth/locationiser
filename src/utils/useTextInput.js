import { useState } from 'react';

export function useTextInput(initialValue) {
  const [value, setValue] = useState(initialValue);

  function onChange(event) {
    setValue(event.currentTarget.value);
  }

  return { value, onChange };
}
