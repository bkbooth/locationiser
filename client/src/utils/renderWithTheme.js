import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components/macro';
import { theme } from 'utils/theme';

function WithThemeProvider({ children }) {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
}

export function renderWithTheme(ui, options) {
  return render(ui, { wrapper: WithThemeProvider, ...options });
}
