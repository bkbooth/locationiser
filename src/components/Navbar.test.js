import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { renderWithTheme } from '../utils/renderWithTheme';
import Navbar from './Navbar';

it('renders correctly', () => {
  const { container } = renderWithTheme(
    <Router>
      <Navbar />
    </Router>
  );
  expect(container.cloneNode(true)).toMatchSnapshot();
});
