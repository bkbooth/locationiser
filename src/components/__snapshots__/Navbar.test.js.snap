// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c4 {
  text-transform: uppercase;
  font-size: 0.7rem;
  -webkit-letter-spacing: 0.05rem;
  -moz-letter-spacing: 0.05rem;
  -ms-letter-spacing: 0.05rem;
  letter-spacing: 0.05rem;
  background: #0094c0;
  color: #fbfbfb;
  padding: 10px 20px;
  border: none;
  border-radius: 5px;
  cursor: pointer;
  -webkit-transition: all 0.2s;
  transition: all 0.2s;
  background: #fbfbfb;
  color: #101010;
  border: 1px solid #656565;
}

.c4:hover,
.c4:focus {
  box-shadow: 1px 1px 10px rgba(0,0,0,0.4);
  outline: none;
}

.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  background: #0094c0;
  color: #fbfbfb;
  padding: 20px;
  box-shadow: 0 5px 10px rgba(0,0,0,0.4);
  z-index: 10;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0;
  font-size: 1.7rem;
  -webkit-letter-spacing: 0.05rem;
  -moz-letter-spacing: 0.05rem;
  -ms-letter-spacing: 0.05rem;
  letter-spacing: 0.05rem;
}

.c1 {
  color: #fbfbfb;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: opacity 0.2s;
  transition: opacity 0.2s;
}

.c1:hover,
.c1:focus {
  opacity: 0.8;
  outline: none;
}

.c3 {
  width: 40px;
  height: auto;
  margin-right: 10px;
}

<div>
  <header
    class="c0"
  >
    <a
      class="c1"
      href="/"
    >
      <h1
        class="c2"
      >
        <img
          alt="world map emoji"
          class="c3"
          src="world-map-emoji.png"
        />
        locations
      </h1>
    </a>
    <nav>
      <button
        class="c4"
        disabled=""
      >
        Login
      </button>
    </nav>
  </header>
</div>
`;
